src/interfaces/hapi.ts

```diff
@@ -37,13 +37,13 @@ import * as hapi from 'hapi';
  * @returns {ErrorMessage} - a partially or fully populated instance of
  *  ErrorMessage
  */
-function hapiErrorHandler(req: hapi.Request, err: {}, config: Configuration) {
+function hapiErrorHandler(req: hapi.Request|undefined, err: {}, config?: Configuration) {
   let service = '';
   let version: string|undefined = '';
 
   if (isObject(config)) {
-    service = config.getServiceContext().service;
-    version = config.getServiceContext().version;
+    service = config!.getServiceContext().service;
```

Comment:
If you change `if (isObject(config))` to just `if (config)`, you wouldn't need the non-null casting. Because the type of `config` is already `Configuration|undefined`, you wouldn't need the `isObject` check.

Reply:
That could be potentially semver major though. We cannot rely on TypeScript type checks as a way of knowing what JS users are actually passing in.

Reply:
The code originally was written in TypeScript and used a lot of `is*` methods to check object types, and I don't want to change that as part of this PR.  However, for new checks I will leverage the TypeScript compiler.